components/TransactionForm.js

```diff
@@ -0,0 +1,346 @@
+              name="category"
+              required
+            >
+              <option value="" disabled>
+                Please select a category
+              </option>
+              {categories.map((category) => (
+                <option key={category.id} value={category.name}>
+                  {category.name}
+                </option>
+              ))}
+            </StyledCategorySelect>
+          </FormRow>
+          <FormRow>
+            <StyledTypeLabel htmlFor="type">Type</StyledTypeLabel>
+            <StyledToggleButton id="type">
+              <StyledRadioInput
+                id="income"
+                name="type"
+                type="radio"
+                value="income"
+                checked={selectedType === "income"}
+                onChange={() => {
+                  setSelectedType("income");
+                  setTypeError(false);
+                }}
+              />
+              <StyledRadioLabel htmlFor="income">Income</StyledRadioLabel>
+
+              <StyledRadioInput
+                id="expense"
+                name="type"
+                type="radio"
+                value="expense"
+                checked={selectedType === "expense"}
+                onChange={() => {
+                  setSelectedType("expense");
+                  setTypeError(false);
+                }}
+              />
+              <StyledRadioLabel htmlFor="expense">Expense</StyledRadioLabel>
+            </StyledToggleButton>
+            {typeError && (
+              <ErrorMessage>Please select a transaction type.</ErrorMessage>
+            )}
+          </FormRow>
+          <FormRow>
+            <StyledDateLabel htmlFor="date">Date</StyledDateLabel>
+            <StyledDateInput
+              type="date"
+              id="date"
+              name="date"
+              defaultValue={today}
+              required
+            ></StyledDateInput>
+          </FormRow>
+        </StyledFieldset>
+
+        <StyledButton type="submit">Add</StyledButton>
+      </StyledForm>
+    </>
+  );
+}
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const StyledFieldset = styled.fieldset`
+  border-radius: 16px;
+  border-width: 1px;
+  border-color: var(--dark-grey-color);
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: repeat(6, auto);
+  grid-template-areas:
+    "nameLabel nameInput"
+    "amountLabel amountInput"
+    "categoryLabel categoryInput"
+    "typeLabel typeToggleButton"
+    ". typeErrorMessage"
+    "dateLabel dateInput";
+  padding: 12px 16px;
+  gap: 4px;
+`;
+
+const FormRow = styled.div`
+  display: contents;
+`;
+
+const StyledNameLabel = styled.label`
+  grid-area: nameLabel;
+`;
+
+const StyledNameInput = styled.input`
+  grid-area: nameInput;
+  padding: 4px 12px;
+  border: 1px solid var(--dark-grey-color);
+  border-radius: 24px;
+  background-color: white;
+  color: #141414;
```

Comment:
Try to be consistent in specifying the color variables. Otherwise you use rgba, so here too.

Reply:
Oh no, I just saw in the `styles.js` that you are using `hex`. Then I would use `hex` everywhere instead of named colors or `rgba`. So here you can leave it as it is! 😁